apps/src/templates/ShareWarningsDialog.jsx

```diff
@@ -0,0 +1,79 @@
+var ShareWarnings = require('./ShareWarnings.jsx');
+
+/**
+ * Modal for our SharingWarnings.
+ */
+var SharingWarningsDialog = module.exports = React.createClass({
+  propTypes: {
+    is13Plus: React.PropTypes.bool.isRequired,
+    showStoreDataAlert: React.PropTypes.bool.isRequired,
+    handleClose: React.PropTypes.func.isRequired,
+    handleTooYoung: React.PropTypes.func.isRequired
+  },
+
+  getInitialState: function() {
+    return { modalIsOpen: !this.props.is13Plus || this.props.showStoreDataAlert };
+  },
+
+  componentDidMount: function () {
+    // We didn't need to show our modal. Go through the close process so that
+    // app becomes unblocked
+    if (!this.state.modalIsOpen) {
+      this.handleClose();
+    }
+  },
+
+  handleClose: function() {
+    this.setState({modalIsOpen: false});
+    React.unmountComponentAtNode(this.getDOMNode().parentNode);
+    this.props.handleClose();
+  },
+
+  render: function () {
+    if (!this.state.modalIsOpen) {
+      return <div/>;
```

Comment:
Struggled to find a lot of info about best practices here. I did however find a comment in this issue https://github.com/facebook/react/issues/1044 stating that unmountComponentAtNode should be used sparingly. I think I'm probably better of not unmounting, and just rendering the empty div when !modalIsOpen.